### DISCUSSION TITLE
Add New Device Error

### DISCUSSION THREAD

**Discussion by ArtHuge** [Koenkk/zigbee2mqtt#21718] [Devices]
I added a new unsupported device and created external definition code, how do I write the configuration file？Thanks

`const {deviceEndpoints, onOff} = require('zigbee-herdsman-converters/lib/modernExtend');

const definition = {
    zigbeeModel: ['UIOT-LNSwitch2[02,02]'],
    model: 'UIOT-LNSwitch2[02,02]',
    vendor: 'UIOT',
    description: 'Automatically generated definition',
    extend: [deviceEndpoints({"endpoints":{"1":1,"2":2,"3":3,"4":4,"5":5}}), deviceEndpoints({"endpoints":{"1":1,"2":2,"3":3,"4":4,"5":5}}), onOff({"powerOnBehavior":false,"endpointNames":["1","2","3","4","5"]})],
    meta: {"multiEndpoint":true},
};

module.exports = definition;`

Log show：
`Debug 2024-03-08 17:09:07Received Zigbee message from '0x00158d0005db24ef', type 'attributeReport', cluster 'genBasic', data '{"32769":{"data":[0,0,0,0],"type":"Buffer"}}' from endpoint 1 with groupID 0

Error 2024-03-08 17:09:15Failed to configure '0x00158d0005db24ef', attempt 3 (Error: Bind 0x00158d0005db24ef/1 genOnOff from '0x00124b0030c3087e/1' failed (AREQ - ZDO - bindRsp after 10000ms) at Object.start (/app/node_modules/zigbee-herdsman/src/utils/waitress.ts:63:23) at /app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:692:28 at processTicksAndRejections (node:internal/process/task_queues:95:5) at Queue.execute (/app/node_modules/zigbee-herdsman/src/utils/queue.ts:35:20) at Endpoint.bind (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:551:13) at setupAttributes (/app/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:56:13) at result.configure (/app/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:125:13) at Object.configure (/app/node_modules/zigbee-herdsman-converters/src/index.ts:141:25) at Configure.configure (/app/lib/extension/configure.ts:122:13) at EventEmitter.wrappedCallback (/app/lib/eventBus.ts:174:17))`

**Comment by chris-1243** [comment]
As soon as your defintion is done, you need to use an external converter as explained in the documentation.

https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html

In `configuration.yaml` add:

```
external_converters:
  - name_of_converter.js
```

Or via the frontend:
Settings
  External converters
    +
      Name_of_converters.js
        Submit

Once added, you will have to restart the application
